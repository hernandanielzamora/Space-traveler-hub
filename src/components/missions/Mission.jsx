import { PropTypes } from 'prop-types';
import React from 'react';

const Mission = ({
  name, description, reserved, applyMission, leaveMission, id,
}) => {
  const badge = reserved ? 'Active Member' : 'Not a member';

  const button = () => {
    if (reserved) {
      return (
        <button onClick={(i) => leaveMission(i)} id={id} className="btn-abandon" type="button">
          Leave Mission
        </button>
      );
    }
    return (
      <button onClick={(i) => applyMission(i)} id={id} className="btn-apply" type="button">
        Join Mission
      </button>
    );
  };

  return (
    <tr className="inner-table">
      <td className="inner-name">{name}</td>
      <td className="inner-description rightBottom-border">{description}</td>
      <td className="rightBottom-border badge">
        <div className={reserved ? 'center-flex active-member' : 'center-flex not-member'}>{badge}</div>
      </td>
      <td className="mission-btnContainer rightBottom-border">{button()}</td>
    </tr>
  );
};

Mission.propTypes = {
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  reserved: PropTypes.bool.isRequired,
  applyMission: PropTypes.func.isRequired,
  leaveMission: PropTypes.func.isRequired,
  id: PropTypes.string.isRequired,
};

export default Mission;
