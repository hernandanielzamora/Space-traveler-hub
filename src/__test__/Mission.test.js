import React from 'react';
import { render } from '@testing-library/react';

import Mission from '../components/missions/Mission';

it('renders Component correctly', () => {
  const tree = render(<Mission />);
  expect(tree).toMatchSnapshot();
});
