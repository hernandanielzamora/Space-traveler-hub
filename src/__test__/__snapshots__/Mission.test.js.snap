// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Component correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <tr
        class="inner-table"
      >
        <td
          class="inner-name"
        />
        <td
          class="inner-description rightBottom-border"
        />
        <td
          class="rightBottom-border badge"
        >
          <div
            class="center-flex not-member"
          >
            Not a member
          </div>
        </td>
        <td
          class="mission-btnContainer rightBottom-border"
        >
          <button
            class="btn-apply"
            type="button"
          >
            Join Mission
          </button>
        </td>
      </tr>
    </div>
  </body>,
  "container": <div>
    <tr
      class="inner-table"
    >
      <td
        class="inner-name"
      />
      <td
        class="inner-description rightBottom-border"
      />
      <td
        class="rightBottom-border badge"
      >
        <div
          class="center-flex not-member"
        >
          Not a member
        </div>
      </td>
      <td
        class="mission-btnContainer rightBottom-border"
      >
        <button
          class="btn-apply"
          type="button"
        >
          Join Mission
        </button>
      </td>
    </tr>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
